### PR TITLE
TF-2925 The message list not showed correctly when change mailbox very quick 

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2925 

## Root cause

- Because `RefreshChangeEmail` executes before `GetAllEmail` but returns the following result. So the email list will not be updated correctly.

## Resolved

[demo.webm](https://github.com/user-attachments/assets/6455ebbf-27ce-4427-ae92-f207ac370ce0)

### CODE REVIEW

**Review by tddang-linagora — APPROVED** [review]

**Comment by tddang-linagora** [comment]
- [x] Test Android OK
- [ ] Test iOS

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 1]
```diff
@@ -586,13 +586,8 @@ class MailboxDashBoardController extends ReloadableController {
   }
 
   void setSelectedMailbox(PresentationMailbox? newPresentationMailbox) {
-    final previousMailbox = selectedMailbox.value;
-    if (previousMailbox == newPresentationMailbox) {
-      selectedMailbox.value = newPresentationMailbox;
-      selectedMailbox.refresh();
```
Comment (hoangdat) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:592]: why we prevent refresh in this case?

**Comment by hoangdat** [comment]
- [ ] we should write a `e2e test` for this issue

File: lib/features/thread/domain/state/get_all_email_state.dart [hunk 2]
```diff
@@ -10,11 +11,20 @@ class GetAllEmailLoading extends LoadingState {}
 class GetAllEmailSuccess extends UIState {
   final List<PresentationEmail> emailList;
   final State? currentEmailState;
+  final MailboxId? currentMailboxId;
```
Comment (hoangdat) [inline lib/features/thread/domain/state/get_all_email_state.dart:14]: How about the case of failure?

File: lib/features/thread/domain/state/refresh_changes_all_email_state.dart [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
 import 'package:jmap_dart_client/jmap/core/state.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
 import 'package:model/email/presentation_email.dart';
 
 class RefreshChangesAllEmailLoading extends LoadingState {}
 
 class RefreshChangesAllEmailSuccess extends UIState {
   final List<PresentationEmail> emailList;
   final State? currentEmailState;
+  final MailboxId? currentMailboxId;
```
Comment (hoangdat) [inline lib/features/thread/domain/state/refresh_changes_all_email_state.dart:12]: how about the case of failure?

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 4]
```diff
@@ -586,13 +586,8 @@ class MailboxDashBoardController extends ReloadableController {
...
-    } else {
-      selectedMailbox.value = newPresentationMailbox;
-    }
+    log('MailboxDashBoardController::setSelectedMailbox: SELECTED_MAILBOX_ID = ${newPresentationMailbox?.id.asString} |  SELECTED_MAILBOX_NAME = ${newPresentationMailbox?.name?.name} | ');
+    selectedMailbox.value = newPresentationMailbox;
```
Comment (hoangdat) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:590]: test with #2694

Comment (hoangdat) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:590]: test with `sending queue`

Comment (hoangdat) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:590]: - [ ] test with #1831

File: lib/features/thread/domain/state/refresh_changes_all_email_state.dart (hunk 3 above)
Comment (dab246) [reply lib/features/thread/domain/state/refresh_changes_all_email_state.dart:12]: > how about the case of failure?

The list will not be displayed

File: lib/features/thread/domain/state/get_all_email_state.dart (hunk 2 above)
Comment (dab246) [reply lib/features/thread/domain/state/get_all_email_state.dart:14]: The list will not be displayed

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart (hunk 1 above)
Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:592]: We've removed the `currentMailboxId` variable from the ThreadView so we won't need to force the mailbox to update again, when it's selected again.

**Comment by dab246** [comment]
> * [ ]  we should write a `e2e test` for this issue

IMO, we should merge PR e2e test before doing it.

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart (hunk 4 above)
Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:590]: > * [x]  test with [System not response correctly view belong to URL once user open email details view then click back button of browser #1831](https://github.com/linagora/tmail-flutter/issues/1831)

Work fine.


https://github.com/user-attachments/assets/eaf61d00-370a-4108-87b7-4f07c4490849

Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:590]: > * [x]  test with `sending queue`

Work fine

https://github.com/user-attachments/assets/26349620-d3ea-4e00-b4d1-aaa280553953

**Review by hoangdat — APPROVED** [review]

**Pull request merged**